jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/jobdetails/JobDetailsPresenter.java

```diff
@@ -0,0 +1,144 @@
+import org.uberfire.client.mvp.PlaceManager;
+import org.uberfire.client.mvp.UberElement;
+import org.uberfire.client.workbench.events.ChangeTitleWidgetEvent;
+import org.uberfire.ext.widgets.common.client.menu.RefreshMenuBuilder;
+import org.uberfire.lifecycle.OnStartup;
+import org.uberfire.mvp.PlaceRequest;
+import org.uberfire.workbench.model.CompassPosition;
+import org.uberfire.workbench.model.Position;
+
+@Dependent
+@WorkbenchScreen(identifier = JobDetailsPresenter.SCREEN_ID, preferredWidth = 655)
+public class JobDetailsPresenter implements RefreshMenuBuilder.SupportsRefresh {
+
+    public static final String SCREEN_ID = "Job Details";
+
+    @Inject
+    public JobDetailsView view;
+
+    @Inject
+    PlaceManager placeManager;
+
+    @Inject
+    private Event<ChangeTitleWidgetEvent> changeTitleWidgetEvent;
+
+    private PlaceRequest place;
+    private String serverTemplateId;
+    private String deploymentId;
+    private Long jobId;
+
+    @Inject
+    private Caller<ExecutorService> executorServices;
+
+    @WorkbenchPartView
+    public UberElement<JobDetailsPresenter> getView() {
+        return view;
+    }
+
+    @DefaultPosition
+    public Position getPosition() {
+        return CompassPosition.EAST;
+    }
+
+    @WorkbenchPartTitle
+    public String getTitle() {
+        return Constants.INSTANCE.Job_Details();
+    }
+
+    @OnStartup
+    public void onStartup(final PlaceRequest place) {
+        this.place = place;
+    }
+
+    public void onJobSelectedEvent(@Observes JobSelectedEvent event) {
+        this.serverTemplateId = event.getServerTemplateId();
+        this.deploymentId = event.getDeploymentId();
+        this.jobId = event.getJobId();
+
+        refreshExecutionErrorDataRemote(serverTemplateId,
+                                        deploymentId,
+                                        jobId);
+    }
+
+    @Override
+    public void onRefresh() {
+        refreshExecutionErrorDataRemote(serverTemplateId,
+                                        deploymentId,
+                                        jobId);
+    }
+
+    public void refreshExecutionErrorDataRemote(final String serverTemplateId,
+                                                final String deploymentId,
+                                                final Long jobId) {
+        executorServices.call(
+                (RequestDetails requestDetails) -> {
+                    view.setValue(requestDetails);
+                    changeTitleWidgetEvent.fire(new ChangeTitleWidgetEvent(this.place,
+                                                                           getJobDetailTitle(requestDetails.getRequest())));
+                })
+                .getRequestDetails(serverTemplateId,
+                                   deploymentId,
+                                   jobId);
+    }
+
+    private String getJobDetailTitle(RequestSummary summary) {
+        String title = summary.getId() + " - ";
```

Comment:
Can you move the extra String part (" - ") to whent key is not null(L120)? That way when key is null only the deployment id is added. This method is probably a good candidate for some isolated testing too, just to check the final String with all the variations.